packages/apex-node/src/logs/logService.ts

```diff
@@ -71,25 +95,82 @@ export class LogService {
     });
   }
 
+  public async getLogById(logId: string): Promise<LogResult> {
+    const baseUrl = this.connection.tooling._baseUrl();
+    const url = `${baseUrl}/sobjects/ApexLog/${logId}/Body`;
+    const response = (await this.connection.tooling.request(url)) as AnyJson;
+    return { log: response.toString() || '' };
+  }
+
   public async getLogRecords(numberOfLogs?: number): Promise<LogRecord[]> {
-    let query = 'Select Id, Application, DurationMilliseconds, Location, ';
-    query +=
-      'LogLength, LogUser.Name, Operation, Request, StartTime, Status from ApexLog Order By StartTime DESC';
+    let apexLogQuery = `
+        SELECT Id, Application, DurationMilliseconds, Location, LogLength, LogUser.Name,
+          Operation, Request, StartTime, Status
+        FROM ApexLog
+        ORDER BY StartTime DESC
+      `;
 
     if (typeof numberOfLogs === 'number') {
       if (numberOfLogs <= 0) {
         throw new Error(nls.localize('numLogsError'));
       }
       numberOfLogs = Math.min(numberOfLogs, MAX_NUM_LOGS);
-      query += ` LIMIT ${numberOfLogs}`;
+      apexLogQuery += ` LIMIT ${numberOfLogs}`;
     }
 
     const response = (await this.connection.tooling.query(
-      query
+      apexLogQuery
     )) as LogQueryResult;
     return response.records as LogRecord[];
   }
 
+  public async tail(tailer?: (log: string) => void): Promise<void> {
+    this.logger = await Logger.child('apexLogApi', { tag: 'tail' });
```

Comment:
could the `org` be provided as a parameter here or can it be derived from the `connection`?

Reply:
Yes, this works for me!

Reply:
And to answer your question, I don't think it's possible to look back up from the `connection` to get the `org`, but passing it in is fine.